elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -312,6 +317,7 @@ public boolean updateAttribute(String fieldName, Object newVal) {
     public boolean updateRelation(String fieldName, Set<PersistentResource> resourceIdentifiers) {
         checkPermission(UpdatePermission.class, this);
         checkFieldPermission(UpdatePermission.class, this, fieldName);
+        checkShareable(resourceIdentifiers);
```

Comment:
This PR isn't complete. I wanted to allow @aklish to look at the logic. Unit tests forthcoming.